synapse/lib/agenda.py

```diff
@@ -922,3 +921,4 @@ async def _runJob(self, user, appt):
                     # fire beholder event before invoking nexus change (in case readonly)
                     await self.core.feedBeholder('cron:stop', {'iden': appt.iden})
                     await self._storeAppt(appt, nexs=True)
+                    await self.core.updateCronStats(appt.iden, appt.pack())
```

Comment:
In this function we're modifying the `appt` directly, then storing a copy of the data in the hive ( `self._storeAppt(...)` ) which is a nexus event, then pushing a new nexus event ( `self.core.updateCronStats()` ) which will make same changes to the `appt` object again [ in the single Cortex case ].

Could we make this a single set of operations + nexus event instead of two events?

Reply:
No, but I could make it reload the appt from hive instead of from the nexus event data. Code updated to reflect this strategy.

Reply:
I think the reload updates must still make it 2 events correct?